src/preproc/preprocessing_pass_registry.h

```diff
@@ -0,0 +1,59 @@
+/*********************                                                        */
+/*! \file preprocessing_pass_registry.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Justin Xu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation for preprocessing pass registry
+ **
+ ** Implementation for preprocessing pass registry, which facilitates
+ *registering passes and deals with destruction of variables like timers of the
```

Comment:
The registry does not destruct the timers itself. The PreprocessingPassRegistry owns the passes and makes sure that they are destroyed when the registry is destroyed. It is also responsible for making the API available to the passes.